hexrdgui/masking/mask_manager_dialog.py

```diff
@@ -0,0 +1,218 @@
+        # write to hdf5
+        with h5py.File(output_file, 'w') as f:
+            h5py_group = f.create_group('masks')
+            MaskManager().write_masks_to_group(data, h5py_group)
+
+    def import_masks(self):
+        selected_file, _ = QFileDialog.getOpenFileName(
+            self.ui, 'Save Mask', HexrdConfig().working_dir,
+            'HDF5 files (*.h5 *.hdf5)')
+
+        if not selected_file:
+            return
+
+        HexrdConfig().working_dir = os.path.dirname(selected_file)
+        # Unwrap the h5 file to a dict
+        masks_dict = {}
+        with h5py.File(selected_file, 'r') as f:
+            unwrap_h5_to_dict(f, masks_dict)
+        MaskManager().load_masks(masks_dict['masks'])
+
+    def masks_to_panel_buffer(self):
+        show_dialog = False
+        selection = 'Replace buffer'
+        for det in HexrdConfig().detectors.values():
+            buff_val = det.get('buffer', {}).get('value', None)
+            if isinstance(buff_val, np.ndarray) and buff_val.ndim == 2:
+                show_dialog = True
+                break
+
+        if show_dialog:
+            dialog = QDialog(self.ui)
+            layout = QVBoxLayout()
+            dialog.setLayout(layout)
+
+            options = QComboBox(dialog)
+            options.addItem('Replace buffer')
+            options.addItem('Logical AND with buffer')
+            options.addItem('Logical OR with buffer')
+            layout.addWidget(options)
+
+            buttons = QDialogButtonBox.Ok | QDialogButtonBox.Cancel
+            button_box = QDialogButtonBox(buttons, dialog)
+            button_box.accepted.connect(dialog.accept)
+            button_box.rejected.connect(dialog.reject)
+            layout.addWidget(button_box)
+
+            if not dialog.exec():
+                # canceled
+                return
+
+            selection = options.currentText()
+
+        MaskManager().masks_to_panel_buffer(selection, buff_val)
+        msg = 'Masks set as panel buffers.'
+        QMessageBox.information(self.parent, 'HEXRD', msg)
+
+    def show_masks(self):
+        num_dets = len(HexrdConfig().detector_names)
+        cols = 2 if num_dets > 1 else 1
+        rows = math.ceil(num_dets / cols)
+
+        fig = plt.figure()
+        fig.canvas.manager.set_window_title('User Created Masks')
+        for i, det in enumerate(HexrdConfig().detector_names):
+            axis = fig.add_subplot(rows, cols, i + 1)
+            axis.set_title(det)
+            axis.imshow(MaskManager().raw_masks_dict[det])
+        fig.canvas.draw_idle()
+        fig.show()
+
+    def update_visibility_checkboxes(self):
+        with block_signals(self.ui.masks_table):
+            for i, key in enumerate(MaskManager().mask_names):
+                cb = self.ui.masks_table.cellWidget(i, 1)
+                status = key in MaskManager().visible_masks
+                cb.setChecked(status)
+        MaskManager().masks_changed()
```

Comment:
Yes, the visibility of masks are changed here. The UI is first updated to reflect the changes and then the function is called so that the appropriate signal is emitted and the images are updated. This function is only called from `hide_all_masks` and `show_all_masks` so the logic here is what was redundant between the two. If you'd prefer though I can move `MaskManager().masks_changed` into `hide_all_masks` and `show_all_masks` instead.

Reply:
Yeah, can you put it in those functions instead? I think clarity is better in this case than redundancy...

Reply:
Sounds good :+1: